xen/arch/arm/coproc/coproc.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * xen/arch/arm/coproc/coproc.c
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include <xen/init.h>
+#include <xen/sched.h>
+#include <xen/list.h>
+#include <asm/device.h>
+
+#include "coproc.h"
+
+static DEFINE_SPINLOCK(coproc_devices_lock);
+static LIST_HEAD(coproc_devices);
+static int num_coprocs_devices;
+
+int vcoproc_register(struct domain *d, struct vcoproc_info *info)
```

Comment:
IMHO, a function with such f name should associate a vcoproc to a coproc device, in other word register a vcoproc within the coproc framework. NOT link the vcoproc to a domain, which is pure vcoproc_domian function